### DISCUSSION TITLE
Adding Thousands Separator to ui.input for Numbers

### DISCUSSION THREAD

**Discussion by o12486vs2** [zauberzeug/nicegui#3483] [Q&A]
### Question

Hello NiceGUI community,
I'm trying to implement a thousands separator (commas) for numeric input in a ui.input component. I've attempted to use the underlying Quasar component properties, but haven't been successful.
Here's what I've tried:
```py
money_format = json.dumps({
    'thousands': ',',
    'precision': 0,
    'masked': False
})
format = f'input-class="text-right" v-money="{money_format}"'
obj = ui.input(value=value,label=item)
obj.props(f'{format} hide-bottom-space dense color="cyan-10" type="{type}"').classes('w-full')
```
This approach doesn't seem to work as expected. The thousands separators are not being applied to the input.
I'm wondering if there's a built-in way to achieve this in NiceGUI, or if anyone has found a reliable method to add thousands separators to numeric inputs.
Any help or suggestions would be greatly appreciated. Thank you!

**Comment by python-and-novella** [comment]
```python3
from nicegui import ui

f = ui.select(
    [" ", ",", "_"],
    value=" ",
    label="Separator:",
    on_change=lambda: bind_input_number.refresh(),
).classes("w-48")
value = 1_000


@ui.refreshable
def bind_input_number():
    with ui.input().props("readonly").classes("w-48") as inp:
        with ui.element("q-popup-edit").props("auto-save").classes("w-48"):
            ui_number = ui.number(
                value=0,
                precision=0,
                on_change=lambda e: inp.set_value(f"{int(e.sender.value):{f.value}}"),
            ).props("dense borderless")
            ui_number.bind_value(globals(), "value")


bind_input_number()

ui.run()
```
![动画](https://github.com/user-attachments/assets/53b92d64-ec0a-4e99-bb49-94edb2a90682)
